lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -378,15 +378,23 @@ role = fromMaybe (error $ "role: unmatched type" <> show (typeRep @c))
 --
 -- > mkAddressPool key g cc (addresses pool) == pool
 addresses
-    :: forall c k. ()
+    :: forall c k. Typeable c
     => (KeyFingerprint "payment" k -> Address)
     -> AddressPool c k
-    -> [(Address, AddressState)]
+    -> [(Address, AddressState, NonEmpty DerivationIndex)]
 addresses mkAddress =
-    map (\(k, (_, st)) -> (mkAddress k, st))
+    map (\(k, (ix, st)) -> (mkAddress k, st, toDerivationPath ix))
     . L.sortOn (fst . snd)
     . Map.toList
     . indexedKeys
+  where
+    toDerivationPath ix = NE.fromList $ map DerivationIndex
+        [ getIndex purposeBIP44
+        , getIndex purposeCIP1852
+        , 0
```

Comment:
Okay-ish because we assume the first account. I'd however rather leave a note as comment explaining this hard-coded value.